http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/nrs/HttpStreamsHandler.java

```diff
@@ -209,11 +213,15 @@ private void handleCancelled(ChannelHandlerContext ctx, InT msg) {
     }
 
     private void handleReadHttpContent(ChannelHandlerContext ctx, HttpContent content) {
+        boolean lastHttpContent = content instanceof LastHttpContent;
+        if (lastHttpContent) {
+            ctx.channel().attr(STREAMING_COMPLETE_KEY).set(true);
+        }
         if (!ignoreBodyRead) {
-            if (content instanceof LastHttpContent) {
-
+            if (lastHttpContent) {
+                logger.debug(ctx.channel(), () -> "Received LastHttpContent " + ctx.channel());
```

Comment:
Maybe moving this log statement to line 218?

Reply:
Done